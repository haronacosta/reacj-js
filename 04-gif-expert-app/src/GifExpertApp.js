import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GridGifs from './components/GridGifs';

const GifExpertApp = () => {
  const initialCategories = ['One Puch'];

  const [categories, setCategories] = useState(initialCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category, i) => (
          <GridGifs key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
