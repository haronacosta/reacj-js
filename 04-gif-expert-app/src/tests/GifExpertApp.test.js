import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas al componente <GifExperApp />', () => {
  test('Se debe de mostrar correctamete el componente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar una lista de categorias', () => {
    const initialCategories = ['Bayern', 'Dortmund'];

    const wrapper = shallow(
      <GifExpertApp initialCategories={initialCategories} />
    );

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('GifGrid').length).toBe(initialCategories.length);
  });
});
