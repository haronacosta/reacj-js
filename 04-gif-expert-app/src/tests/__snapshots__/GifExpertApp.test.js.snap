// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifExperApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Bayern"
      key="Bayern"
    />
    <GifGrid
      category="Dortmund"
      key="Dortmund"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas al componente <GifExperApp /> Se debe de mostrar correctamete el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
